DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/ip_authoring_mmhost/README.md

```diff
@@ -0,0 +1,368 @@
+
+| Signal | Description | Width
+| --- | --- | ---
+| `address` | Represents the byte address | 1 - 64 bits
+| `writedata` | Data for write transfers | 8, 16, 32, 64, 128, 256, 512, or 1024 bits
+| `readdata` | The readdata driven from the agent to the host in response to a read transfer | 8, 16, 32, 64, 128, 256, 512, or 1024 bits
+| `byteenable` | Enables one or more specific byte lanes during transfers on interfaces of width greater than 8 bits. Each bit in byteenable corresponds to a byte in writedata and readadata | 2, 4, 8, 16, 32, 64, or 128 bits
+| `waitrequest` | An agent asserts this waitrequest signal when unable to respond to read/write requests | 1 bit 
+| `readdatavalid` | Used for variable-latency pipelined read transfers, asserts that the readdata signal contains valid data | 1 bit
+| `burstcount` | Used to indicate the number of transfers in each burst | 1 - 11 bits
+
+### Buffer Location in Pointer Address Bits
+The compiler encodes certain information regarding the virtual address space in top bits of the 64 bits wide pointer address. Below is a table showing which bits correspond to what function
+
+| Bits | Function
+| --- | ---
+| 0 - 40 | Addressing within the memory system
+| 41 - 46 | Buffer Location information
+| 47 | Not Used
+| 48 - 61 | Used by runtime layer
+| 62 - 63 | Indicates whether a pointer is a global or local memory pointer
+
+Users do not need to encode this information themselves in most cases. The compiler automatically generates logic to embed this information from the buffer location specified on the pointer kernel argument in the source file.
+
+Why is this embedding needed?
+
+In some cases, the compiler cannot figure out which buffer location a pointer corresponds to, and it creates logic in the generated RTL that inspects the top bits of the pointer at runtime to detect the buffer location and route the memory transaction to the correct global memory
+
+
+
+### Tutorial Code Overview
+This tutorial demonstrates how to implement a memory-mapped interface in an IP authoring flow. The design performs a multiply and add operation on two input vectors, stores the result of the multiple in the second vector, and the result of the add in an output-only vector.
+
+* Memory-mapped interfaces needs to contain all parameters to work correctly.
+
+## Testing the Tutorial
+In ```mmhost.cpp```, two seperate Kernal IP's are declared for the same operation of adding and multiplying two vectors. ```PointerIP``` which declares a global memory interface for all pointers, and ```VectorMADIP``` which declares a seperate interface buffer location to each argument. 
+```c++
+struct PointerIP {
+  int *x, *y, *z;
+  int size;
+  
+  PointerIP(int *x_, int *y_, int *z_, int size_)
```

Comment:
Remove the constructor from these examples here. We just want to highlight how the argument interfaces are specified here.